builtin/game/chatcommands.lua

```diff
@@ -451,59 +457,130 @@ core.register_chatcommand("set", {
 	end,
 })
 
-local function emergeblocks_callback(pos, action, num_calls_remaining, ctx)
-	if ctx.total_blocks == 0 then
-		ctx.total_blocks   = num_calls_remaining + 1
-		ctx.current_blocks = 0
+local function emerge_callback(pos, action, num_calls_remaining, ctx)
+	if ctx.total_count == 0 then
+		ctx.total_count   = num_calls_remaining + 1
+		ctx.current_count = 0
 	end
-	ctx.current_blocks = ctx.current_blocks + 1
-
-	if ctx.current_blocks == ctx.total_blocks then
+	ctx.current_count = ctx.current_count + 1
+	if core.get_us_time() < ctx.cur_time then
+		-- core.get_us_time() loops after 2^32 microseconds.
+		-- The possibility of two loops having occurred (at least
+		-- 72 minutes have elapsed) is ignored deliberately as
+		-- it's unlikely, and non-critical anyway.
+		ctx.start_time = ctx.start_time - 2^32
```

Comment:
Recomputing it is a bit useless indeed. Although I assume luajit (if used) will optimize it...

Nevertheless, I'm inclined towards keeping 2^32, as it is more readable. `0x100000000` is more zeroes than a casual reader can absorb at a glance. And the code is rarely executed and not really performance-critical. If necessary, I could precompute 2^32, but I'm not sure that it's worth the trouble.